pagerduty/resource_pagerduty_event_orchestration_path_router_test.go

```diff
@@ -0,0 +1,403 @@
+					resource.TestCheckResourceAttr(
+						"pagerduty_event_orchestration_router.router", "sets.0.rules.1.conditions.0.expression", "event.severity matches part 'critical'"),
+				),
+			},
+			{
+				Config: testAccCheckPagerDutyEventOrchestrationPathConfigWithCatchAllToService(team, escalation_policy, service, orchestration),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyEventOrchestrationPathExists("pagerduty_event_orchestration_router.router"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_event_orchestration_router.router", "sets.0.rules.#", "1"),
+					testAccCheckPagerDutyEventOrchestrationPathRouteToMatch(
+						"pagerduty_event_orchestration_router.router", "pagerduty_service.bar", true), //test for catch_all routing to service if provided
+				),
+			},
+			{
+				Config: testAccCheckPagerDutyEventOrchestrationPathConfigDeleteAllRulesInSet(team, escalation_policy, service, orchestration),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyEventOrchestrationPathExists("pagerduty_event_orchestration_router.router"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_event_orchestration_router.router", "sets.0.rules.#", "0"),
+					testAccCheckPagerDutyEventOrchestrationPathRouteToMatch(
+						"pagerduty_event_orchestration_router.router", "pagerduty_service.bar", true),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckPagerDutyEventOrchestrationPathDestroy(s *terraform.State) error {
+	client, _ := testAccProvider.Meta().(*Config).Client()
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "pagerduty_event_orchestration_path_router" {
+			continue
+		}
+
+		orch, _ := s.RootModule().Resources["pagerduty_event_orchestration.orch"]
+
+		if _, _, err := client.EventOrchestrationPaths.Get(orch.Primary.ID, "router"); err == nil {
+			return fmt.Errorf("Event Orchestration Path still exists")
+		}
+	}
+	return nil
+}
+
+func testAccCheckPagerDutyEventOrchestrationPathExists(rn string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("Not found: %s", rn)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Event Orchestration Path Type is set")
+		}
+
+		orch, _ := s.RootModule().Resources["pagerduty_event_orchestration.orch"]
+		client, _ := testAccProvider.Meta().(*Config).Client()
+		found, _, err := client.EventOrchestrationPaths.Get(orch.Primary.ID, "router")
+
+		if err != nil {
+			return fmt.Errorf("Orchestration Path type not found: %v for orchestration %v", "router", orch.Primary.ID)
+		}
+		if found.Type != "router" {
+			return fmt.Errorf("Event Orchrestration path not found: %v - %v", "router", found)
+		}
+		return nil
+	}
+}
+
+func createBaseConfig(t, ep, s, o string) string {
+	return fmt.Sprintf(`
+	resource "pagerduty_team" "foo" {
+		name = "%s"
+	}
+
+	resource "pagerduty_user" "foo" {
+		name        = "user"
+		email       = "user@pagerduty.com"
```

Comment:
Should the user have the name prefixed with `tf-` in order for the sweeper to delete it?